From f187175ff1715e5780db2f9b597fc6d20abfbe66 Mon Sep 17 00:00:00 2001
From: Chris Gorman <chrisjohgorman@gmail.com>
Date: Tue, 13 Aug 2024 13:59:42 -0400
Subject: [PATCH] gr_modtool: util_functions.py add newline to variable
 modified_last

I would like someone to confirm that this fix works for linux.  Under
windows this fixes a problem with python modifying __init__.py imporperly
and I believe this will occur under linux as well, but I have not tested
with linux.

I would also like to ensure that this doesn't negatively impact cpp code
generation as I have not yet tested it with cpp.

Signed-off-by: Chris Gorman <chrisjohgorman@gmail.com>
---
 gr-utils/modtool/tools/util_functions.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gr-utils/modtool/tools/util_functions.py b/gr-utils/modtool/tools/util_functions.py
index 6bab3249474..99a6bde4406 100644
--- a/gr-utils/modtool/tools/util_functions.py
+++ b/gr-utils/modtool/tools/util_functions.py
@@ -42,7 +42,7 @@ def append_re_line_sequence(filename: str, linepattern: str, newline: str, closi
     if closing_parentheses is not None and last_line.rstrip().endswith(closing_parentheses):
         modified_last = f'\n{last_line.rstrip()[:-len(closing_parentheses)]}\n{closing_parentheses}\n'
     else:
-        modified_last = last_line + '\n'
+        modified_last = last_line + newline + '\n'
     newfile = oldfile.replace(last_line, modified_last)
     with open(filename, 'w') as f:
         f.write(newfile)
